Task 1, 1st PR

instanceof-inf opened the pull request.

instanceof-inf committed:
Task 1, 1st PR

MrCaseys commented on src/main/java/com/johan/blignaut/entelect/vitalityincubator/task1/ApplicationConfiguration.java:
```diff
@@ -0,0 +1,18 @@
+import org.springframework.stereotype.Component;
+
+@Component
+public class ApplicationConfiguration {
```
incorrect annotation here, this is your configuration class, this is where you should be defining your beans and dependencies

please check the difference between @bean, @Configration and @Componant, where to use.

MrCaseys commented on src/main/java/com/johan/blignaut/entelect/vitalityincubator/task1/HelloWorldController.java:
```diff
@@ -0,0 +1,32 @@
+@RestController
+public class HelloWorldController {
+
+    private ApplicationConfiguration applicationConfiguration;
```
I believe you're trying to inject your message here, please create a bean in your configuration class and inject it here instead the entire configuration

MrCaseys commented:
Hey @instanceof-inf 🙂 , @Component cannot act like a @Configuration. @Configuration is used for defining beans and their dependencies, and @Component is used for general-purpose auto-detection of components.

and remember, your aim is to inject your message..not the configuration, later on, you'll see that your configuration class will have many more beans defined there...so you don't want to inject everything everywhere.

instanceof-inf committed:
Task 1, 1st PR update.

Removed @Component annotations and added correct @Configuration and @Bean annotations

RossVanDerHeyde commented on src/main/java/com/johan/blignaut/entelect/vitalityincubator/task1/HelloWorldController.java:
```diff
@@ -0,0 +1,24 @@
+public class HelloWorldController {
+
+    @Autowired
+    private HelloWorld helloWorld;
```
Please use constuctor injection instead of field injection.
It seems like a trivial change, but constructor injection makes unit testing much easier.

instanceof-inf committed:
Changed HelloWorld injection from field to contructor.
